### PR TITLE
Bluetooth: Clean up RSCS and DDFS services and samples

### PR DESCRIPTION
**Pull request by KAGA164**
Fix typos in source code and documentation. This also improves the services and samples logic.

### CODE REVIEW

File: samples/bluetooth/peripheral_rscs/src/main.c [hunk 1]
```diff
@@ -64,15 +84,15 @@ int set_cumulative(uint32_t total_distance)
...
+int calibration(void)
 {
 	printk("Start the calibration procedure\n");
 	return 0;
 }
 
 void update_loc(uint8_t location)
 {
-	printk("Update sensor location: %d\n", location);
+	printk("Update sensor location: %u - %s\n", location, sensor_location[location]);
```
Comment (dchat-nordic) [inline samples/bluetooth/peripheral_rscs/src/main.c:95]: What if `location` is larger than size of `sensor_location`?

**Review by peknis — APPROVED** [review]

File: subsys/bluetooth/services/ddfs.c [hunk 2]
```diff
@@ -277,23 +280,24 @@ static void ctrl_point_ind(struct bt_conn *conn, uint8_t req_op, uint8_t status,
...
 		ddfs_inst.indicating = true;
 	}
 }
 
 int bt_ddfs_distance_measurement_notify(struct bt_conn *conn,
 					const struct bt_ddfs_distance_measurement *measurement)
 {
-	if (!ddfs_inst.dm_notification_enabled) {
-		return -EACCES;
-	}
```
Comment (maje-emb) [inline subsys/bluetooth/services/ddfs.c:296]: Why don't we want to check if notification is enabled?

File: subsys/bluetooth/services/ddfs.c [hunk 3]
```diff
@@ -277,23 +280,24 @@ static void ctrl_point_ind(struct bt_conn *conn, uint8_t req_op, uint8_t status,
...
+	static const struct bt_gatt_attr *attr = &ddf_svc.attrs[DDF_SVC_DISTANCE_MEAS_ATTR_IDX];
```
Comment (maje-emb) [inline subsys/bluetooth/services/ddfs.c:300]: Can we use `bt_gatt_find_by_uuid`? 
Then `DDF_SVC_DISTANCE_MEAS_ATTR_IDX` will be unnecessary.
Also in other places.

File: subsys/bluetooth/services/ddfs.c [hunk 4]
```diff
@@ -51,15 +51,11 @@ LOG_MODULE_REGISTER(bt_ddfs, CONFIG_BT_DDFS_LOG_LEVEL);
...
-#define DDF_SVC_CTRL_POINT_ATTR_IDX	12
 
 static struct {
 	struct bt_gatt_indicate_params ind_params;
 	struct bt_ddfs_features dm_features;
 	const struct bt_ddfs_cb *cb;
-	bool dm_notification_enabled;
-	bool am_notification_enabled;
-	bool em_notification_enebled;
 	bool dm_ctrl_pt_indicate_enabled;
```
Comment (maje-emb) [inline subsys/bluetooth/services/ddfs.c:59]: The variable is not used.

File: samples/bluetooth/nrf_dm/README.rst [hunk 5]
```diff
@@ -61,7 +61,7 @@ This message contains the following fields:
    * Identifier indicating that the device supports distance measurement.
    * The access address used for measurement.
 
-This scanner/advertiser interaction is used as a synchronisation point.
+This scanner/advertiser interaction is used as a synchronization point.
```
Comment (maje-emb) [inline samples/bluetooth/nrf_dm/README.rst:64]: btw I noticed that both versions are used in the repository.

File: subsys/bluetooth/services/ddfs.c (hunk 2 above)
Comment (KAGA164) [reply subsys/bluetooth/services/ddfs.c:296]: I do check just before sending notification by calling `bt_gatt_is_subscribed`

File: subsys/bluetooth/services/ddfs.c (hunk 4 above)
Comment (KAGA164) [reply subsys/bluetooth/services/ddfs.c:59]: Well spotted. Removed

File: samples/bluetooth/nrf_dm/README.rst (hunk 5 above)
Comment (KAGA164) [reply samples/bluetooth/nrf_dm/README.rst:64]: You are right. Can we go with current version?

Comment (peknis) [reply samples/bluetooth/nrf_dm/README.rst:64]: Needs to be with z

File: subsys/bluetooth/services/ddfs.c (hunk 3 above)
Comment (KAGA164) [reply subsys/bluetooth/services/ddfs.c:300]: I wanted to use it but I don't know a call context of this function so there could be a race when two threads would execute this function first time, both thread could call then `bt_gatt_find_by_uuid` but the result will be the same even then. I decided to hardcode attr in the API functions. What do you think?

File: samples/bluetooth/peripheral_rscs/src/main.c (hunk 1 above)
Comment (KAGA164) [reply samples/bluetooth/peripheral_rscs/src/main.c:95]: Added the assertion for checking this

**Review by dchat-nordic — APPROVED** [review]

File: subsys/bluetooth/services/ddfs.c (hunk 3 above)
Comment (maje-emb) [reply subsys/bluetooth/services/ddfs.c:299]: Yes, race can occur at the first check. So hardcoded attr seems to be a better option.

**Review by maje-emb — APPROVED** [review]

**Pull request merged**